### PR TITLE
Add cauchy kernel cookbook

### PR DESCRIPTION
**Pull request by vinx13**

### CODE REVIEW

**Comment by karlnapf** [comment]
Thanks for this!

As said a few times before, let's not have more kernel cookbooks (after this one) as we already have tons of them. Rather, we should have cookbooks for algorithms that haven't been described yet

File: doc/cookbook/source/examples/kernel/cauchy.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
+=============
+Cauchy Kernel
+=============
+
+The cauchy kernel is derived from the Cauchy distribution :cite:`basak2008kernel`. It is a long-tailed kernel defined as
```
Comment (karlnapf) [inline doc/cookbook/source/examples/kernel/cauchy.rst]: Minor: pls start new sentences in a new line

File: doc/cookbook/source/examples/kernel/cauchy.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+.. math::
+  K(x,x') = \frac{1}{1+\frac{\| x-x' \|^2}{\sigma}}
```
Comment (karlnapf) [inline doc/cookbook/source/examples/kernel/cauchy.rst]: I would prefer a lowercase k, unless K is used elsewhere

File: doc/cookbook/source/index.rst [hunk 3]
```diff
@@ -96,6 +96,16 @@ Distance
 
    examples/distance/**
 
+Kernel
```
Comment (karlnapf) [inline doc/cookbook/source/index.rst]: oh wait, we didnt have kernels yet? ;)
Ok then this is quite good to have :)

File: doc/cookbook/source/references.bib [hunk 4]
```diff
@@ -163,3 +163,10 @@ @article{hyvarinen2000independent
   pages={411--430},
   year={2000}
 }
+@inproceedings{basak2008kernel
+	author={J. Basak},
+	booktitle={2008 19th International Conference on Pattern Recognition},
```
Comment (karlnapf) [inline doc/cookbook/source/references.bib]: is this really the best reference? where do you have it from? can't we use some of the references we already used (like the kernel book by schölkopf?)

File: doc/cookbook/source/references.bib [hunk 5]
```diff
@@ -163,3 +163,10 @@ @article{hyvarinen2000independent
...
+	title={A least square kernel machine with box constraints},
+	year={2008},
+	pages={1-4},
```
Comment (karlnapf) [inline doc/cookbook/source/references.bib]: pls do proper references. These pages here are definitely not correct

File: doc/cookbook/source/examples/kernel/cauchy.rst [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+-------
+Example
+-------
+Imagine we have files with data. We create :sgclass:`CDenseFeatures` (here 64 bit floats aka RealFeatures) as
+
+.. sgexample:: cauchy:create_features
+
+We create an instance of :sgclass:`CCauchyKernel`.
+
+.. sgexample:: cauchy:create_kernel
```
Comment (karlnapf) [inline doc/cookbook/source/examples/kernel/cauchy.rst]: Could you talk about the fact that a distance is used for this kernel?

File: doc/cookbook/source/references.bib (hunk 4 above)
Comment (vinx13) [reply doc/cookbook/source/references.bib]: i found several links point to this paper, e.g https://github.com/gmum/pykernels/blob/c8afdc79a15197ad3be2a0db0118f5e948577f49/pykernels/regular.py#L132
and I've checked that paper, the author gave a definition of it in sec.3

File: doc/cookbook/source/references.bib (hunk 5 above)
Comment (vinx13) [reply doc/cookbook/source/references.bib]: ICPR papers are not re-assigned a page number when they are published. This is based on the reference format provided in https://ieeexplore.ieee.org/document/4761717

Comment (karlnapf) [reply doc/cookbook/source/references.bib]: I see, maybe we can remove them then?

File: doc/cookbook/source/references.bib (hunk 4 above)
Comment (karlnapf) [reply doc/cookbook/source/references.bib]: ok cool. When citing it, put [Section 3] in there (if possible)

**Comment by karlnapf** [comment]
I restarted windows since it was failing

File: doc/cookbook/source/references.bib [hunk 7]
```diff
@@ -167,6 +167,5 @@ @inproceedings{basak2008kernel
 	author={J. Basak},
 	booktitle={2008 19th International Conference on Pattern Recognition},
 	title={A least square kernel machine with box constraints},
-	year={2008},
-	pages={1-4},
+	year={2008}
```
Comment (karlnapf) [inline doc/cookbook/source/references.bib]: cool thx!

Did you render this locally (the html output) and check that it all works ?

Comment (vinx13) [reply doc/cookbook/source/references.bib]: Yes it works

**Comment by karlnapf** [comment]
Alright, great! thanks!

**Pull request merged**